x-pack/plugins/observability_ai_assistant/public/hooks/use_timeline.ts

```diff
@@ -194,7 +195,12 @@ export function useTimeline({
 
   return {
     items,
-    onEdit: (item, content) => {},
+    onEdit: async (item, newMessage) => {
+      const index = messages.findIndex((message) => message['@timestamp'] === item['@timestamp']);
```

Comment:
Needed an id to find the original message in the array, and `id` wasn't part of Message, but `@timestamp` wasn't part of `ChatTimelineItem`.

I think it would be preferable we generate `id` higher up than in `get_timeline_items_from_conversation`, perhaps in `useTimeline`.